pkg/daemon/update.go

```diff
@@ -440,6 +446,91 @@ func (dn *Daemon) applyOSChanges(oldConfig, newConfig *mcfgv1.MachineConfig) (re
 
 }
 
+func calculatePostConfigChangeActionFromFileDiffs(oldIgnConfig, newIgnConfig ign3types.Config) (actions []string) {
+	filesPostConfigChangeActionNone := []string{
+		"/var/lib/kubelet/config.json",
+	}
+	filesPostConfigChangeActionReloadCrio := []string{
+		"/etc/containers/registries.conf",
+	}
+
+	oldFileSet := make(map[string]ign3types.File)
+	for _, f := range oldIgnConfig.Storage.Files {
+		oldFileSet[f.Path] = f
+	}
+	newFileSet := make(map[string]ign3types.File)
+	for _, f := range newIgnConfig.Storage.Files {
+		newFileSet[f.Path] = f
+	}
+	diffFileSet := []string{}
+
+	// First check if any files were removed
+	for path := range oldFileSet {
+		_, ok := newFileSet[path]
+		if !ok {
+			// debug: remove
+			glog.Infof("File diff: %v was deleted", path)
+			diffFileSet = append(diffFileSet, path)
+		}
+	}
+
+	// Now check if any files were added/changed
+	for path, newFile := range newFileSet {
+		oldFile, ok := oldFileSet[path]
+		if !ok {
+			// debug: remove
+			glog.Infof("File diff: %v was added", path)
+			diffFileSet = append(diffFileSet, path)
+		}
+		if !reflect.DeepEqual(oldFile, newFile) {
```

Comment:
This should run only when file was found in both old and new MC.

Reply:
ah good catch! this works as it today I think but will cause a duplicate add of a file. Will fix